lib/custom_lib/custom_lib.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2021, Legrand North America, LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* The C preprocessor namespace is global, so Zephyr regularly
+ * uses encoded pathnames in the header file multiple-inclusion guards.
+ * For example:
+ *
+ *   zephyr/lib/libc/minimal/include/sys/timespec.h uses
+ *   ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_SYS_TIMESPEC_H_
```

Comment:
this is something that is widely known by C developers, so I think it doesn't add much value

Reply:
Accepted, and manually rolled in.